altair/vegalite/v5/api.py

```diff
@@ -913,10 +933,23 @@ def condition(
 # Top-level objects
 
 
+def _top_schema_base(obj: Any, /):  # -> <subclass of SchemaBase and TopLevelMixin>
```

Comment:
This is similar to a recent change in `utils.core.py` 

I haven't seen this behaviour documented for either `mypy` or `pyright`, but until https://github.com/python/typing/issues/213 has been formalised - this is the best we can do without changing the class hierarchy

https://github.com/vega/altair/blob/9afd374a83f954e79a5fd6e4f8a8a9c4e5381c04/altair/utils/core.py#L855-L876

Reply:
Wasn't aware of intersections being discussed, that's going to be very useful with the various mixins used in the Altair code base :) Nice find!